Steps/UploadCodeCoverage.yml

```diff
@@ -0,0 +1,77 @@
+## @file
+# Azure Pipelines step template upload code coverage to codecov.io
+#
+# Follows the codecov.io documentation for uploading code coverage reports:
+# https://docs.codecov.com/docs/codecov-uploader
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+parameters:
+- name: codecov_token
+  displayName: Codecov.io Token
+  type: string
+  default: ''
+- name: report_dir
+  displayName: Code Coverage Report
+  type: string
+  default: ''
+
+steps:
+- task: PythonScript@0
+  displayName: Download and Verify Codecov Uploader
+  inputs:
+    scriptSource: inline
+    script: |
+      import platform
+      import requests
+      import hashlib
+      import os
+
+      system = platform.system()
+
+      if system == 'Windows':
+        url = 'https://uploader.codecov.io/latest/windows/codecov.exe'
+        filename = 'codecov.exe'
+        checksum_url = 'https://uploader.codecov.io/latest/windows/codecov.exe.SHA256SUM'
+        checksum_filename = 'codecov.exe.SHA256SUM'
+        print(f'##vso[task.setvariable variable=codecov_uploader_cmd].\{filename}')
+      elif system == 'Linux':
+        url = 'https://uploader.codecov.io/latest/linux/codecov'
+        filename = 'codecov'
+        checksum_url = 'https://uploader.codecov.io/latest/linux/codecov.SHA256SUM'
+        checksum_filename = 'codecov.SHA256SUM'
+        print(f'##vso[task.setvariable variable=codecov_uploader_cmd]./{filename}')
+      else:
+        print(f'##[error]Unsupported Host System! System = {system}.')
+        print(f'##vso[task.complete result=Failed;]Unsupported Host System! System = {system}.')
+
+      response = requests.get(url)
+      if response.status_code == 200:
```

Comment:
If the response getting the application is not successful, is there a point in continuing to checksum verification?

Reply:
Similar for the other response below.

Reply:
Added the print statements to change the result to failed as requested when the request response was not 200. Also added the particular status code to the statement.